compiler/noirc_frontend/src/hir_def/types/unification.rs

```diff
@@ -0,0 +1,893 @@
+        if let Type::InfixExpr(lhs_lhs, lhs_op, lhs_rhs, _) = self {
+            let lhs_op_inverse = lhs_op.inverse();
+
+            // Check if it's `A + rhs = other` or `A - rhs = other`
+            if let (Some(op_a_inverse), Type::TypeVariable(lhs_lhs_var)) =
+                (lhs_op_inverse, lhs_lhs.as_ref())
+            {
+                if lhs_lhs_var.1.borrow().is_unbound() {
+                    // We can say that `A = other - rhs` or `A = other + rhs` respectively
+                    let new_rhs =
+                        Type::infix_expr(Box::new(other.clone()), op_a_inverse, lhs_rhs.clone());
+
+                    let mut tmp_bindings = bindings.clone();
+                    if lhs_lhs.try_unify(&new_rhs, &mut tmp_bindings).is_ok() {
+                        *bindings = tmp_bindings;
+                        return Ok(());
+                    }
+                }
+            }
+
+            // Check if it's `lhs + B = other`
+            if let (BinaryTypeOperator::Addition, Type::TypeVariable(lhs_rhs_var)) =
+                (lhs_op, lhs_rhs.as_ref())
+            {
+                if lhs_rhs_var.1.borrow().is_unbound() {
+                    // We can say that `B = other - lhs`
+                    let new_rhs = Type::infix_expr(
+                        Box::new(other.clone()),
+                        BinaryTypeOperator::Subtraction,
+                        lhs_lhs.clone(),
+                    );
+
+                    let mut tmp_bindings = bindings.clone();
+                    if lhs_rhs.try_unify(&new_rhs, &mut tmp_bindings).is_ok() {
+                        *bindings = tmp_bindings;
+                        return Ok(());
+                    }
+                }
+            }
+
+            // Check if it's `lhs - B = other`
+            if let (BinaryTypeOperator::Subtraction, Type::TypeVariable(lhs_rhs_var)) =
+                (lhs_op, lhs_rhs.as_ref())
+            {
+                if lhs_rhs_var.1.borrow().is_unbound() {
+                    // We can say that `B = lhs - other`
+                    let new_rhs = Type::infix_expr(
+                        lhs_lhs.clone(),
+                        BinaryTypeOperator::Subtraction,
+                        Box::new(other.clone()),
+                    );
+
+                    let mut tmp_bindings = bindings.clone();
+                    if lhs_rhs.try_unify(&new_rhs, &mut tmp_bindings).is_ok() {
+                        *bindings = tmp_bindings;
+                        return Ok(());
+                    }
+                }
+            }
+        }
+
+        Err(UnificationError)
+    }
+
+    /// Try to unify the following equations:
+    /// - `(..a..) + 1 = (..b..)` -> `(..a..) = (..b..) - 1`
+    /// - `(..a..) - 1 = (..b..)` -> `(..a..) = (..b..) + 1`
+    /// - `(..a..) = (..b..) + 1` -> `(..b..) = (..a..) - 1`
+    /// - `(..a..) = (..b..) - 1` -> `(..b..) = (..a..) + 1`
+    fn try_unify_by_moving_single_constant_term(
```

Comment:
This seems very similar to `try_unify_by_moving_constant_terms` - from the comments above both they seem identical - what is the difference? Do we need both?